### PR TITLE
rpc: allow RPC requests on GET too

### PR DESCRIPTION
**Pull request by karalabe**
This PR simply makes the HTTP RPC server respond to GET requests too as it seems most client code was written like that.

**Note, although this does fix the issue, do we really want to support GET? What clients are using GET? Anything that's JavaScript will default to POST.**

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Tue Feb 2 17:30:55 UTC 2016_

**Comment by obscuren** [comment]
@karalabe Please remove this as well https://github.com/ethereum/go-ethereum/pull/2169/files#diff-7a7d022a3310612a978fecd722fd4e21R169

**Comment by karalabe** [comment]
That's why it was panicing... I noticed it once but was just wondering where it came from...

**Comment by karalabe** [comment]
Fixed, rebased and force pushed.

**Comment by obscuren** [comment]
Yup. That's what I mentioned during the stand up. I wanted to PR this as well but figured you might as well do this in this PR.

**Comment by obscuren** [comment]
:+1:

**Comment by obscuren** [comment]
To answer your question, we don't know, that's what makes it difficult to just swap it out. I know some people use raw `curl` and `GET`.

**Comment by karalabe** [comment]
curl does POST by default if you supply a form `-d`

File: rpc/http.go [hunk 1]
```diff
@@ -212,17 +206,17 @@ func httpTimestamp(t time.Time) []byte {
 	t = t.UTC()
 	yy, mm, dd := t.Date()
 	hh, mn, ss := t.Clock()
-	day := days[3 * t.Weekday():]
```
Comment (fjl) [inline rpc/http.go]: Not directly related to this PR, but why does this function even exist?
The only use of this function can basically be replaced by `time.Now().Format(http.TimeFormat)`.

**Comment by fjl** [comment]
:+1:

File: rpc/http.go (hunk 1 above)
Comment (fjl) [inline rpc/http.go]: I am puzzled about the whole thing actually, why are we reimplementing the HTTP server?

**Pull request merged**